sim/src/main.rs

```diff
@@ -38,11 +36,11 @@ fn main() {
     // Create simulator object.
     let mut simulator: Simulator = Simulator::new();
 
-    let tgen = simulator.add_node(TrafficGenerator::new(1, 0));
-    let node1 = simulator.add_node(Link::new(2, 1, plugin_str, 1));
-    let node2 = simulator.add_node(Link::new(2, 1, plugin_str, 2));
-    let node3 = simulator.add_node(Link::new(2, 1, plugin_str, 3));
-    let node4 = simulator.add_node(Link::new(2, 1, plugin_str, 4));
+    let tgen = simulator.add_node(Node::new(2, 10, 1, plugin_str, 0)); // traffic generator
```

Comment:
I do not think that abstraction works particularly well. Maybe this might need some rethinking. What if a node has multiple output links with different capacities, and egress rates? This seems fairly confusing to read now.

Reply:
I changed the abstraction to just take in the information needed for the object.  Now the user only needs to know the functions in simulator.rs, which should make it more intuitive to use.